cat-rekognition/src/handler.js

```diff
@@ -0,0 +1,49 @@
+'use strict';
+
+const AWS = require('aws-sdk');
+
+const rekognition = new AWS.Rekognition();
+
+module.exports.imgClassification = (event, context, callback) => {
+
+    console.log('Received event:', JSON.stringify(event, null, 2));
+
+    const filesToBeChecked = this.recordsToFiles(this.filterEvents(event));
+
+    filesToBeChecked.forEach(function (fileName) {
+        var params = {
+            Image: {
+                S3Object: {
+                    Bucket: "cat-detector-img-repo",
+                    Name: fileName
+                }
+            },
+            MaxLabels: 10,
+            MinConfidence: 60
+        };
+
+        rekognition.detectLabels(params, function(err, data) {
+            if (err) console.log(err, err.stack);
+            else     console.log(data);
+        });
+    });
+
+    callback(null, event);
+
+};
+
+module.exports.filterEvents = (s3Events) => {
+    const records = s3Events['Records'];
+    const fileAddedEvents = records.filter(function (event) {
+        return event['eventName'] == 'ObjectCreated:Put'
+    });
+    return fileAddedEvents;
+};
+
```

Comment:
nitpicking: inconsistent number of lines between functions - here we have 2 above only 1